util/ArrayOfLength.ts

```diff
@@ -0,0 +1,7 @@
+export type ArrayOfLength<
+  T,
+  L extends number,
+  A extends T[] = [],
+> = number extends L ? T[]
+  : L extends A["length"] ? A
+  : ArrayOfLength<T, L, [...A, T]>
```

Comment:
@harrysolovay Would this be the right place for this utility type?

Reply:
Sure :)